python_modules/dagster/dagster/_model/decorator.py

```diff
@@ -41,21 +44,43 @@ def _namedtuple_model_transform(
         * bans tuple methods that don't make sense for a model object
         * creates a run time checked __new__  (optional).
     """
-    base = NamedTuple(f"_{cls.__name__}", cls.__annotations__.items())
+    field_set = getattr(cls, "__annotations__", {})
+    defaults = {name: getattr(cls, name) for name in field_set.keys() if hasattr(cls, name)}
+
+    base = NamedTuple(f"_{cls.__name__}", field_set.items())
     nt_new = base.__new__
     if checked:
+        eval_ctx = EvalContext.capture_from_frame(
+            1 + decorator_frames,
+            add_to_local_ns={_DEFAULT_VALUES_KEY: defaults},
```

Comment:
Yeah this is the starting to get to "magical incantations" territory here with the hardcoded keys and the like.